x-pack/plugins/security_solution/public/plugin.tsx

```diff
@@ -465,65 +462,47 @@ export class Plugin implements IPlugin<PluginSetup, PluginStart, SetupPlugins, S
    * Register deepLinks and appUpdater for all app links, to change deepLinks as needed when licensing changes.
    */
   async registerAppLinks(core: CoreStart, plugins: StartPlugins) {
-    licenseService.start(plugins.licensing.license$);
-    const licensing = licenseService.getLicenseInformation$();
-
-    const newNavEnabled = core.uiSettings.get(ENABLE_GROUPED_NAVIGATION, false);
-    if (newNavEnabled) {
-      registerDeepLinksUpdater(this.appUpdater$);
-    }
-
     const { links, getFilteredLinks } = await this.lazyApplicationLinks();
 
-    const linksPermissions: LinksPermissions = {
-      experimentalFeatures: this.experimentalFeatures,
-      capabilities: core.application.capabilities,
-    };
+    const { license$ } = plugins.licensing;
+    const newNavEnabled$ = core.uiSettings.get$<boolean>(ENABLE_GROUPED_NAVIGATION, false);
 
-    if (licensing == null) {
-      // update without license (defaults to "basic")
-      updateAppLinks(links, linksPermissions);
+    let appLinksSubscription: Subscription | null = null;
+    license$.pipe(combineLatestWith(newNavEnabled$)).subscribe(async ([license, newNavEnabled]) => {
```

Comment:
combine `license$` and  `newNavEnabled$` advanced setting observables, if any of _license_ or _newNavEnabled_ changes the callback is executed